RELEASING.md

```diff
@@ -0,0 +1,12 @@
+# How to release a new version
```

Comment:
Great to see a `RELEASING.md` guide! I think it would be useful to include a link to the Go module version numbering documentation https://go.dev/doc/modules/version-numbers